modules/core/js/src/main/scala/coursier/core/compatibility/package.scala

```diff
@@ -34,10 +44,10 @@ package object compatibility {
   lazy val DOMParser = newFromXmlDomOrGlobal("DOMParser")
   lazy val XMLSerializer = newFromXmlDomOrGlobal("XMLSerializer")
   lazy val sax =
-    if (js.isUndefined(g.sax))
-      g.require("sax")
+    if (js.isUndefined(globalObject.sax))
+      globalObject.require("sax")
     else
-      g.sax
+      globalObject.sax
```

Comment:
If you do things this way, you'll *still* have the problem that "`require` is not defined". You're going out of your way to reproduce the behavior of Scala.js 0.6.x, which is the cause of #1618.

You should take full advantage of the behavior of Scala.js 1.x instead, as follows:
```scala
lazy val sax =
  if (js.typeOf(g.sax) == "undefined")
    g.require("sax")
  else
    g.sax
```
Likewise, you should inline `newFromXmlDomOrGlobal` at the two places where it's used to be able to use the same pattern:
```scala
lazy val DOMParser = {
  val defn =
    if (js.typeOf(g.DOMParser) == "undefined") g.require("xmldom").DOMParser
    else g.DOMParser
  js.Dynamic.newInstance(defn)()
}
```
and copy-paste for `XMLSerializer`.

If you can't stand the duplication, you can abstract away *some* of it as follows:
```scala
def newFromXmlDomOrGlobal(typeOf: String, globalDefn: => js.Dynamic, xmldomName: String) = {
  val defn =
    if (typeOf == "undefined") g.require("xmldom").selectDynamic(xmldomName)
    else globalDefn
  js.Dynamic.newInstance(defn)()
}
```
which you can use as
```scala
lazy val DOMParser = newFromXmlDomOrGlobal(js.typeOf(g.DOMParser), g.DOMParser, "DOMParser")
lazy val XMLSerializer = newFromXmlDomOrGlobal(js.typeOf(g.XMLSerializer), g.XMLSerializer, "XMLSerializer")
```